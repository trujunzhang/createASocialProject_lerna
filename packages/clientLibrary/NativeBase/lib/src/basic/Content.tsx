import * as React from 'react'
import * as ReactNative from 'react-native'
import * as PropTypes from 'prop-types'
import { KeyboardAwareScrollView } from '@app/react-native-keyboard-aware-scroll-view'
import { connectStyle } from '@app/native-base-shoutem-theme'
import mapPropsToStyleNames from '../utils/mapPropsToStyleNames'
import { platformVariables as variable } from '@app/native-base-variables'
import { isIphoneX } from '@app/react-native-iphone-x-helper'

export interface IContentProps {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle> | any
  /**
     * The theme prop can be applied to any component of NativeBase.
     */
  refreshing?: boolean
  refreshControl?: object | any
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean | any
  enableResetScrollToCoords?: boolean
  contentOffset?: Object | any
  scrollEnabled?: boolean
  contentContainerStyle?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
  keyboardShouldPersistTaps?: string | any
  keyboardDismissMode?: string | any
}
class Content extends React.Component<IContentProps, any> {
  private _root: any
  private _scrollview: any
  static contextTypes = {
    theme: PropTypes.object
  }
  constructor(props) {
    super(props)
    this.state = {
      orientation: 'portrait'
    }
  }
  layoutChange(val) {
    let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight)
    if (val.width >= maxComp) this.setState({ orientation: 'landscape' })
    else {
      this.setState({ orientation: 'portrait' })
    }
  }

  calculateLeft(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let leftPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingLeft !== undefined
      ) {
        leftPadder =
          (this.props.style[1].paddingLeft
            ? this.props.style[1].paddingLeft
            : this.props.style[1].padding) + InsetValues.leftInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingLeft !== undefined
    ) {
      leftPadder =
        (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
        InsetValues.leftInset
    } else {
      leftPadder = InsetValues.leftInset
    }
    return leftPadder
  }
  calculateRight(mode, inSet) {
    let inset: any = null
    if (inSet !== undefined) {
      inset = inSet
    } else {
      inset = variable.Inset
    }
    const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape
    let rightPadder = null
    if (this.props.style[1] !== undefined) {
      if (
        this.props.style[1].padding !== undefined ||
        this.props.style[1].paddingRight !== undefined
      ) {
        rightPadder =
          (this.props.style[1].paddingRight
            ? this.props.style[1].paddingRight
            : this.props.style[1].padding) + InsetValues.rightInset
      }
    } else if (
      this.props.style.padding !== undefined ||
      this.props.style.paddingRight !== undefined
    ) {
      rightPadder =
        (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
        InsetValues.rightInset
    } else {
      rightPadder = InsetValues.rightInset
    }
    return rightPadder
  }
  render() {
    const variables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable
    return isIphoneX() ? (
      <KeyboardAwareScrollView
        automaticallyAdjustContentInsets={false}
        resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
        keyboardShouldPersistTaps={
          this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
        }
        ref={(c) => {
          this._scrollview = c
          this._root = c
        }}
        {...this.props}
        onLayout={(e) => this.layoutChange(e.nativeEvent.layout)}
        style={[
          this.props.style,
          {
            paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
          }
        ]}
        contentContainerStyle={[
          { padding: this.props.padder ? variables.contentPadding : undefined },
          this.props.contentContainerStyle
        ]}>
        {this.props.children}
      </KeyboardAwareScrollView>
    ) : (
        <KeyboardAwareScrollView
          automaticallyAdjustContentInsets={false}
          resetScrollToCoords={(this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }) as any}
          keyboardShouldPersistTaps={
            this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled'
          }
          ref={(c) => {
            this._scrollview = c
            this._root = c
          }}
          {...this.props}
          contentContainerStyle={[
            { padding: this.props.padder ? variables.contentPadding : undefined },
            this.props.contentContainerStyle
          ]}>
          {this.props.children}
        </KeyboardAwareScrollView>
      )
  }
}

// Content.propTypes = {
//     style: PropTypes.oneOfType([
//         PropTypes.object,
//         PropTypes.number,
//         PropTypes.array
//     ]),
//     padder: PropTypes.bool,
//     disableKBDismissScroll: PropTypes.bool,
//     enableResetScrollToCoords: PropTypes.bool,
//     keyboardShouldPersistTaps: PropTypes.string
// };

const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content)

export { StyledContent as Content }
