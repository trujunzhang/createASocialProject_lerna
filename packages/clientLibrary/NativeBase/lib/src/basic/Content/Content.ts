import * as React from 'react'
import { ReactNative, RnTextStyleProp, RnViewStyleProp } from '@appUtils/rnCommonProps'

/**
 * see Widget Content.js
 */
export interface ContentProps {
  /**
   * The theme prop can be applied to any component of NativeBase.
   */
  refreshing?: boolean
  refreshControl?: object
  theme?: Object
  padder?: boolean
  disableKBDismissScroll?: boolean
  enableResetScrollToCoords?: boolean
  contentOffset?: Object
  scrollEnabled?: boolean
  style?: RnViewStyleProp | Array<RnViewStyleProp>
  contentContainerStyle?: RnViewStyleProp | Array<RnViewStyleProp>
  keyboardShouldPersistTaps?: string
  keyboardDismissMode?: string
}
