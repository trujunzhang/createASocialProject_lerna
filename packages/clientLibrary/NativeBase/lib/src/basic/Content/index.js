import * as React from 'react';
import * as PropTypes from 'prop-types';
// TODO:DJZHANG(NB)
import { getKeyboardAwareScrollView } from '@appUtils/cacheHelper';
import { isIphoneX } from 'react-primitives';
// import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view'
import { connectStyle } from 'native-base-shoutem-theme';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
import { mapPropsToStyleNames } from '@app/tools';
class Content extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            orientation: 'portrait'
        };
    }
    layoutChange(val) {
        let maxComp = Math.max(variable.deviceWidth, variable.deviceHeight);
        if (val.width >= maxComp)
            this.setState({ orientation: 'landscape' });
        else {
            this.setState({ orientation: 'portrait' });
        }
    }
    calculateLeft(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let leftPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingLeft !== undefined) {
                leftPadder =
                    (this.props.style[1].paddingLeft
                        ? this.props.style[1].paddingLeft
                        : this.props.style[1].padding) + InsetValues.leftInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingLeft !== undefined) {
            leftPadder =
                (this.props.style.paddingLeft ? this.props.style.paddingLeft : this.props.style.padding) +
                    InsetValues.leftInset;
        }
        else {
            leftPadder = InsetValues.leftInset;
        }
        return leftPadder;
    }
    calculateRight(mode, inSet) {
        let inset = null;
        if (inSet !== undefined) {
            inset = inSet;
        }
        else {
            inset = variable.Inset;
        }
        const InsetValues = mode === 'portrait' ? inset.portrait : inset.landscape;
        let rightPadder = null;
        if (this.props.style[1] !== undefined) {
            if (this.props.style[1].padding !== undefined ||
                this.props.style[1].paddingRight !== undefined) {
                rightPadder =
                    (this.props.style[1].paddingRight
                        ? this.props.style[1].paddingRight
                        : this.props.style[1].padding) + InsetValues.rightInset;
            }
        }
        else if (this.props.style.padding !== undefined ||
            this.props.style.paddingRight !== undefined) {
            rightPadder =
                (this.props.style.paddingRight ? this.props.style.paddingRight : this.props.style.padding) +
                    InsetValues.rightInset;
        }
        else {
            rightPadder = InsetValues.rightInset;
        }
        return rightPadder;
    }
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        const KeyboardAwareScrollView = getKeyboardAwareScrollView();
        return isIphoneX() ? (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { onLayout: (e) => this.layoutChange(e.nativeEvent.layout), style: [
                this.props.style,
                {
                    paddingLeft: this.calculateLeft(this.state.orientation, variables.Inset)
                }
            ], contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children)) : (React.createElement(KeyboardAwareScrollView, Object.assign({ name: 'keyboardAwareScrollView', automaticallyAdjustContentInsets: false, resetScrollToCoords: this.props.disableKBDismissScroll ? null : { x: 0, y: 0 }, keyboardShouldPersistTaps: this.props.keyboardShouldPersistTaps ? this.props.keyboardShouldPersistTaps : 'handled', ref: (c) => {
                this._scrollview = c;
                this._root = c;
            } }, this.props, { contentContainerStyle: [
                { padding: this.props.padder ? variables.contentPadding : undefined },
                this.props.contentContainerStyle
            ] }), this.props.children));
    }
}
Content.contextTypes = {
    theme: PropTypes.object
};
// Content.propTypes = {
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   padder: PropTypes.bool,
//   disableKBDismissScroll: PropTypes.bool,
//   enableResetScrollToCoords: PropTypes.bool,
//   keyboardShouldPersistTaps: PropTypes.string
// }
const StyledContent = connectStyle('NativeBase.Content', {}, mapPropsToStyleNames)(Content);
const TabContent = StyledContent;
export { StyledContent as Content, TabContent };
//# sourceMappingURL=index.js.map