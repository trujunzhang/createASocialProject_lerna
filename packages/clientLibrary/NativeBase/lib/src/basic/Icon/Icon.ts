import * as React from 'react'

/**
 * see Widget Icon.js
 */
export interface IconProps {
  name?: string
  type?:
    | 'AntDesign'
    | 'Entypo'
    | 'EvilIcons'
    | 'Feather'
    | 'FontAwesome'
    | 'FontAwesome5'
    | 'Foundation'
    | 'Ionicons'
    | 'MaterialCommunityIcons'
    | 'MaterialIcons'
    | 'Octicons'
    | 'SimpleLineIcons'
    | 'Zocial'
  // TODO position attribute of ReactNative.FlexStyle hasn't another position values without "absolute" and "relative"
  style?: any
  onPress?: (e?: any) => any
  active?: boolean
  ios?: string
  android?: string
  color?: string
  fontSize?: number
  // Others
  iconSize?: number
  iconColor?: string
}
