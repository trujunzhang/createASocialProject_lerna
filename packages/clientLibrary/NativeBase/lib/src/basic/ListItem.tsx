import * as React from 'react'
import * as ReactNative from 'react-native';
import {
  TouchableHighlight,
  Platform,
  TouchableNativeFeedback,
  View
} from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";
import { platformVariables as variable } from '@app/native-base-variables'

export interface IListItemProps extends TouchableHighlight {
  style?: ReactNative.ViewStyle | Array<ReactNative.ViewStyle>
}
class ListItem extends React.Component<IListItemProps, any> {
  private _root: any
  static contextTypes = {
    theme: PropTypes.object
  };
  render() {
    const variables = this.context.theme
      ? this.context.theme["@@shoutem.theme/themeStyle"].variables
      : variable;

    if (
      Platform.OS === "ios" ||
      Platform.OS === "web" ||
      variables.androidRipple === false ||
      (!this.props.onPress && !this.props.onLongPress) ||
      Platform.Version <= 21
    ) {
      return (
        <TouchableHighlight
          onPress={this.props.onPress}
          onLongPress={this.props.onLongPress}
          ref={c => (this._root = c)}
          underlayColor={variables.listBtnUnderlayColor}
          {...this.props}
          style={this.props.touchableHighlightStyle}
        >
          <View {...this.props}>{this.props.children}</View>
        </TouchableHighlight>
      );
    } else {
      return (
        <TouchableNativeFeedback
          ref={c => (this._root = c)}
          {...this.props}
        >
          <View style={{ marginLeft: -17, paddingLeft: 17 }}>
            <View {...this.props}>{this.props.children}</View>
          </View>
        </TouchableNativeFeedback>
      );
    }
  }
}

// ListItem.propTypes = {
//   ...TouchableHighlight.propTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ]),
//   touchableHighlightStyle: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.array
//   ]),
//   itemDivider: PropTypes.bool,
//   button: PropTypes.bool
// };

const StyledListItem = connectStyle(
  "NativeBase.ListItem",
  {},
  mapPropsToStyleNames
)(ListItem);

export { StyledListItem as ListItem };
