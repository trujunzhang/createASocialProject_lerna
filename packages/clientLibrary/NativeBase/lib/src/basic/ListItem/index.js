import * as React from 'react';
import * as PropTypes from 'prop-types';
import { TouchableHighlight, Platform, TouchableNativeFeedback, View } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { mapPropsToStyleNames } from '@app/tools';
import { platformVariables as variable } from '@appVariables/theme/variables/platform'; // variables // [variable]
class ListItem extends React.Component {
    render() {
        const variables = this.context.theme
            ? this.context.theme['@@shoutem.theme/themeStyle'].variables
            : variable;
        if (Platform.OS === 'ios' ||
            Platform.OS === 'web' ||
            variables.androidRipple === false ||
            (!this.props.onPress && !this.props.onLongPress) ||
            Platform.Version <= 21) {
            return (React.createElement(TouchableHighlight, Object.assign({ onPress: this.props.onPress, onLongPress: this.props.onLongPress, ref: (c) => (this._root = c), underlayColor: variables.listBtnUnderlayColor }, this.props, { style: this.props.touchableHighlightStyle }),
                React.createElement(View, Object.assign({}, this.props, { testID: undefined }), this.props.children)));
        }
        else {
            return (React.createElement(TouchableNativeFeedback, Object.assign({ ref: (c) => (this._root = c) }, this.props),
                React.createElement(View, { style: { marginLeft: -17, paddingLeft: 17 } },
                    React.createElement(View, Object.assign({}, this.props, { testID: undefined }), this.props.children))));
        }
    }
}
ListItem.contextTypes = {
    theme: PropTypes.object
};
// ListItem.propTypes = {
//   ...TouchableHighlight.propTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   touchableHighlightStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   itemDivider: PropTypes.bool,
//   button: PropTypes.bool
// }
const StyledListItem = connectStyle('NativeBase.ListItem', {}, mapPropsToStyleNames)(ListItem);
export { StyledListItem as ListItem };
//# sourceMappingURL=index.js.map