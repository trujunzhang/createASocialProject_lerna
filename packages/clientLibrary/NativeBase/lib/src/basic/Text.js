import * as React from 'react'
import PropTypes from "prop-types";
import { Text as RNText } from "react-primitives";
import { LodashUtils as _ } from '@app/tools'

import { connectStyle } from "@app/native-base-shoutem-theme";
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Text extends React.Component {
  render() {
    const {
      uppercase,
      children,
    } = this.props;

    let text;
    if (uppercase) {
      text = React.Children.map(children, child => {
        if (_.isString(child)) {
          return _.toUpper(child);
        } else {
          return child;
        }
      })
    } else {
      text = children;
    }

    return (
      <RNText ref={c => (this._root = c)} {...this.props}>
        {text}
      </RNText>
    );
  }
}

Text.propTypes = {
  ...RNText.propTypes,
  uppercase: PropTypes.bool,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
};

Text.defaultProps = {
  uppercase: false,
};

const StyledText = connectStyle("NativeBase.Text", {}, mapPropsToStyleNames)(Text);

export { StyledText as Text };
