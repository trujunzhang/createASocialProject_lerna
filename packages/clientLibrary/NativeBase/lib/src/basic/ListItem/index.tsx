import * as React from 'react'
import * as PropTypes from 'prop-types'
import { TouchableHighlight, Platform, TouchableNativeFeedback, View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'
import { mapPropsToStyleNames } from '@app/tools'
import { IThemeVariables } from '@app/tools' // typings
import { platformVariables as variable } from '@appVariables/theme/variables/platform' // variables // [variable]

import { ListItemProps } from './ListItem'

class ListItem extends React.Component<ListItemProps, any> {
  static contextTypes = {
    theme: PropTypes.object
  }
  private _root: any
  render() {
    const variables: IThemeVariables = this.context.theme
      ? this.context.theme['@@shoutem.theme/themeStyle'].variables
      : variable

    if (
      Platform.OS === 'ios' ||
      Platform.OS === 'web' ||
      variables.androidRipple === false ||
      (!this.props.onPress && !this.props.onLongPress) ||
      Platform.Version <= 21
    ) {
      return (
        <TouchableHighlight
          onPress={this.props.onPress}
          onLongPress={this.props.onLongPress}
          ref={(c) => (this._root = c)}
          underlayColor={variables.listBtnUnderlayColor}
          {...this.props}
          style={this.props.touchableHighlightStyle}>
          <View {...this.props} testID={undefined}>
            {this.props.children}
          </View>
        </TouchableHighlight>
      )
    } else {
      return (
        <TouchableNativeFeedback ref={(c) => (this._root = c)} {...this.props}>
          <View style={{ marginLeft: -17, paddingLeft: 17 }}>
            <View {...this.props} testID={undefined}>
              {this.props.children}
            </View>
          </View>
        </TouchableNativeFeedback>
      )
    }
  }
}

// ListItem.propTypes = {
//   ...TouchableHighlight.propTypes,
//   style: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array]),
//   touchableHighlightStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   itemDivider: PropTypes.bool,
//   button: PropTypes.bool
// }

const StyledListItem = connectStyle('NativeBase.ListItem', {}, mapPropsToStyleNames)(ListItem)

export { StyledListItem as ListItem }
