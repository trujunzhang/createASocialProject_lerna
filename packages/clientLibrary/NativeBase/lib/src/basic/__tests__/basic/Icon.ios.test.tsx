import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Content } from '../../Content'
import { ListItem } from '../../ListItem'
import { Icon } from '../../Icon'
import { Text } from '../../Text'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//  return Platform;
// });

it('renders Icon and Icons with icon family', () => {
  const tree = renderer
    .create(
      <Content padder>
        <ListItem>
          <Icon name="logo-apple" style={{ width: 45, height: 45 }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Ionicons</Text>
          <Icon type="Ionicons" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Entypo</Text>
          <Icon type="Entypo" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Feather</Text>
          <Icon type="Feather" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>FontAwesome</Text>
          <Icon type="FontAwesome" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Foundation</Text>
          <Icon type="Foundation" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Material Icons</Text>
          <Icon type="MaterialIcons" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Material Community Icons</Text>
          <Icon type="MaterialCommunityIcons" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Octicons</Text>
          <Icon type="Octicons" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Zocial</Text>
          <Icon type="Zocial" name="dropbox" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Simple Line Icons</Text>
          <Icon type="SimpleLineIcons" name="home" style={{ color: '#999' }} />
        </ListItem>
        <ListItem>
          <Text style={{ width: 220 }}>Evil Icons</Text>
          <Icon type="EvilIcons" name="trophy" style={{ color: '#999' }} />
        </ListItem>
      </Content>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
