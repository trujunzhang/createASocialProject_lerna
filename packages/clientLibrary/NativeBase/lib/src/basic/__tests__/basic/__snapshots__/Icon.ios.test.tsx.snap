// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Icon and Icons with icon family 1`] = `
<view
  name="keyboardAwareScrollView"
  style={Object {}}
>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          logo-apple
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Ionicons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          ios-home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Entypo
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Feather
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          FontAwesome
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Foundation
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Material Icons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Material Community Icons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Octicons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Zocial
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          dropbox
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Simple Line Icons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          home
        </text>
      </view>
    </view>
  </view>
  <view
    name="View"
  >
    <view
      name="View"
      style={Object {}}
    >
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          Evil Icons
        </text>
      </view>
      <view
        name="View"
        style={Object {}}
      >
        <text
          style={Object {}}
        >
          trophy
        </text>
      </view>
    </view>
  </view>
</view>
`;
